File: 0002. Add Two Numbers/Solution.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: Optional[ListNode]
        :type l2: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        
        res = ListNode(0)
        r = res
        h1 = l1
        h2 = l2 

        carry = 0
        add = 0
        a,b = 0,0

        while h1 or h2 or carry:
            a = h1.val if h1 else 0
            b = h2.val if h2 else 0
            
            add = a+b+carry
            carry = add/10

            r.next = ListNode(add%10)

            if h1:
                h1 = h1.next
            if h2:
                h2 = h2.next

            r = r.next

        return res.next

File: 0002. Add Two Numbers/Solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        ListNode res = new ListNode(0);
        ListNode r = res;
        ListNode h1 = l1;
        ListNode h2 = l2;
        int carry = 0;
        int sum = 0;
        int a,b;

        while (h1!=null || h2!=null || carry!=0){
            a = h1!=null?h1.val:0;
            b = h2!=null?h2.val:0;

            sum = a+b+carry;
            carry = sum/10;

            r.next = new ListNode(sum%10);

            if(h1!=null){
                h1 = h1.next;
            }
            if(h2!=null){
                h2 = h2.next;
            }

            r = r.next;
        }
        return res.next;
    }
}